addon/services/prefetch.js

```diff
@@ -28,11 +28,22 @@ if (gte('3.6.0')) {
                 let { route, fullParams } = changeSet.for[i];
                 if (seenRoutes.has(route)) continue;
 
-                  route._prefetched = new RSVP.Promise(r => {
-                    return r(route.prefetch(fullParams, transition))
+                let isRoutePromiseLike = route && route.then instanceof Function && route.catch instanceof Function;
```

Comment:
let me know if you have a preferred convention of checking if something is a promise

Reply:
I think I would do something like:

```js
function isThenable(obj) {
  if (typeof obj === 'object' && obj !== null || typeof obj === 'function') {
    return typeof obj.then === 'function';
  }

  return false;
}
```